### PR TITLE
Fix a NullReferenceException in Signature help

### PR DESCRIPTION
**Pull request by Cosifne**
Fix https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1675669
TryGetSelectedIndex could only throw NRE if the ImmutableArray is not initialized.
image

And I looked at the call stack, this looks like caused because we call the constructor of ImmutableArray.

So if there is no possible memberGroup, NRE would be hit
Fix is quite simple and straightforward.

### CODE REVIEW

File: src/Features/VisualBasic/Portable/SignatureHelp/InvocationExpressionSignatureHelpProvider.vb [hunk 1]
```diff
@@ -116,12 +116,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.SignatureHelp
...
-            Dim accessibleMembers = New ImmutableArray(Of ISymbol)
-            If memberGroup.Length > 0 Then
-                accessibleMembers = GetAccessibleMembers(invocationExpression, semanticModel, within, memberGroup, cancellationToken)
-                items.AddRange(GetMemberGroupItems(accessibleMembers, document, invocationExpression, semanticModel))
+            Dim accessibleMembers = ImmutableArray(Of ISymbol).Empty
+            If memberGroup.Length = 0 Then
+                Return Nothing
             End If
 
+            accessibleMembers = GetAccessibleMembers(invocationExpression, semanticModel, within, memberGroup, cancellationToken)
```
Comment (Youssef1313) [inline src/Features/VisualBasic/Portable/SignatureHelp/InvocationExpressionSignatureHelpProvider.vb:124]:
```suggestion
            If memberGroup.Length = 0 Then
                Return Nothing
            End If

            Dim accessibleMembers = GetAccessibleMembers(invocationExpression, semanticModel, within, memberGroup, cancellationToken)
```

File: src/Features/VisualBasic/Portable/SignatureHelp/InvocationExpressionSignatureHelpProvider.vb [hunk 2]
```diff
@@ -116,12 +116,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.SignatureHelp
...
 
             Dim items = New List(Of SignatureHelpItem)
-            Dim accessibleMembers = New ImmutableArray(Of ISymbol)
-            If memberGroup.Length > 0 Then
-                accessibleMembers = GetAccessibleMembers(invocationExpression, semanticModel, within, memberGroup, cancellationToken)
-                items.AddRange(GetMemberGroupItems(accessibleMembers, document, invocationExpression, semanticModel))
+            Dim accessibleMembers = ImmutableArray(Of ISymbol).Empty
+            If memberGroup.Length = 0 Then
+                Return Nothing
             End If
```
Comment (Youssef1313) [inline src/Features/VisualBasic/Portable/SignatureHelp/InvocationExpressionSignatureHelpProvider.vb:122]: Should this condition be moved few lines up (before or after the `memberGroup = memberGroup.Sort(semanticModel, invocationExpression.SpanStart)` line)?

Comment (Cosifne) [reply src/Features/VisualBasic/Portable/SignatureHelp/InvocationExpressionSignatureHelpProvider.vb]: Done

File: src/Features/VisualBasic/Portable/SignatureHelp/InvocationExpressionSignatureHelpProvider.vb [hunk 3]
```diff
@@ -116,11 +119,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.SignatureHelp
...
 
             Dim items = New List(Of SignatureHelpItem)
-            Dim accessibleMembers = New ImmutableArray(Of ISymbol)
-            If memberGroup.Length > 0 Then
-                accessibleMembers = GetAccessibleMembers(invocationExpression, semanticModel, within, memberGroup, cancellationToken)
-                items.AddRange(GetMemberGroupItems(accessibleMembers, document, invocationExpression, semanticModel))
-            End If
+            Dim accessibleMembers = ImmutableArray(Of ISymbol).Empty
+
+            accessibleMembers = GetAccessibleMembers(invocationExpression, semanticModel, within, memberGroup, cancellationToken)
```
Comment (Youssef1313) [inline src/Features/VisualBasic/Portable/SignatureHelp/InvocationExpressionSignatureHelpProvider.vb:124]:
```suggestion
            Dim accessibleMembers As ImmutableArray(Of ISymbol) = GetAccessibleMembers(invocationExpression, semanticModel, within, memberGroup, cancellationToken)
```

**Comment by jasonmalinowski** [comment]
Can we add a test?

**Review by genlu — APPROVED** [review]
Could you please add a test?

**Comment by Cosifne** [comment]
> Can we add a test?

It might not be easy to know the case triggered this without a dump. And I just have stack trace..

**Pull request merged**